File: app/src/main/java/com/example/myapplication/View/WireHeadphonesFragment.java
package com.example.myapplication.View;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.Adapter.CatalogAdapter;
import com.example.myapplication.Other.VerticalSpacesItemDecoration;
import com.example.myapplication.ViewModel.WireHeadphonesViewModel;
import com.example.myapplication.databinding.FragmentWireheadphonesBinding;
import java.util.ArrayList;

import Models.linked.Product;

public class WireHeadphonesFragment extends Fragment implements LifecycleOwner {

    private FragmentWireheadphonesBinding binding;
    RecyclerView WireHeadphoneRecView;
    WireHeadphonesViewModel viewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentWireheadphonesBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        WireHeadphoneRecView = binding.WireRecView;

        VerticalSpacesItemDecoration SpacesItemDecoration = new VerticalSpacesItemDecoration(15);
        WireHeadphoneRecView.addItemDecoration(SpacesItemDecoration);

        viewModel = new ViewModelProvider(this).get(WireHeadphonesViewModel.class);
        viewModel.getUserMutableLiveData().observe((LifecycleOwner) getContext(), updateWireHeadphones);

        return root;
    }
    Observer<ArrayList<Product>> updateWireHeadphones = new Observer<ArrayList<Product>>() {
        @Override
        public void onChanged(ArrayList<Product> products) {
            CatalogAdapter ProductAdapter = new CatalogAdapter(products);
            WireHeadphoneRecView.setAdapter(ProductAdapter);
        }
    };
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/myapplication/View/WireHeadphonesFragment.kt
package com.example.myapplication.View

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.LifecycleOwner
import Models.linked.Product
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import com.example.myapplication.Adapter.CatalogAdapter
import com.example.myapplication.Other.VerticalSpacesItemDecoration
import com.example.myapplication.ViewModel.WireHeadphonesViewModel
import com.example.myapplication.databinding.FragmentWireheadphonesBinding
import org.koin.androidx.viewmodel.ext.android.viewModel
import java.util.ArrayList

class WireHeadphonesFragment : Fragment(), LifecycleOwner {
    private var binding: FragmentWireheadphonesBinding? = null
    lateinit var WireHeadphoneRecView: RecyclerView
    val viewModel: WireHeadphonesViewModel by viewModel()
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {

        binding = FragmentWireheadphonesBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        WireHeadphoneRecView = binding!!.WireRecView

        val SpacesItemDecoration = VerticalSpacesItemDecoration(15)
        WireHeadphoneRecView.addItemDecoration(SpacesItemDecoration)

        viewModel.WireHeadListLiveData.observe((context as LifecycleOwner?)!!, updateWireHeadphones)
        viewModel.getWireHead(0, 5)

        return root
    }

    var updateWireHeadphones: Observer<ArrayList<Product>?> =
        Observer<ArrayList<Product>?> { products ->
            val ProductAdapter = CatalogAdapter(products)
            WireHeadphoneRecView.adapter = ProductAdapter
        }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}